File: sample.rb
#!/usr/bin/env ruby

def catalan(n)
    cat = Array.new(n+1, 0)
    cat[0] = 1
    (1..n).each do |i|
      (0...i).each do |j|
        cat[i] += cat[j] * cat[i-j-1]
      end
    end
    cat[n]
  end
  
  def main
    N = 30
    start_time = Time.now
    (0...N).each do |i|
      puts "C(#{i}) = #{catalan(i)}"
    end
    end_time = Time.now
    elapsed_ms = (end_time - start_time) * 1000
    puts "Tiempo de ejecución (en Ruby): %.2f ms" % elapsed_ms
  end
  
  main

File: sample.py
#!/usr/bin/env python3

import time


def catalan(n):
    cat = [0] * (n+1)
    cat[0] = 1
    for i in range(1, n+1):
        for j in range(i):
            cat[i] += cat[j] * cat[i-j-1]
    return cat[n]


def main():
    N = 30  # Calculamos C(0) hasta C(29)
    start_time = time.time()
    for i in range(N):
        print(f"C({i}) = {catalan(i)}")
    end_time = time.time()
    elapsed_ms = (end_time - start_time) * 1000
    print(f"Tiempo de ejecución (en Python): {elapsed_ms:.2f} ms")


if __name__ == "__main__":
    main()
